### PR TITLE
4diac integration

### PR DESCRIPTION
**Pull request by Ruslan227**

### CODE REVIEW

File: code/4diac-integration/src/main/kotlin/org/fbme/ide/integration/fordiac/lua/BasicFBTypeLuaTranslator.kt [hunk 1]
```diff
@@ -0,0 +1,668 @@
...
+
+object BasicFBTypeLuaTranslator {
+    private const val FB_DI_FLAG = 33554432
+    private const val FB_DO_FLAG = 67108864
+    private const val FB_AEO_FLAG = 134217728
+    private const val FB_ADI_FLAG = 167772160
+    private const val FB_ADO_FLAG = 201326592
+    private const val FB_IN_FLAG = 268435456
+    private const val FB_AEI_FLAG = 805306368
+    private const val NEW_LINE: String = "\n"
```
Comment (qradimir) [inline code/4diac-integration/src/main/kotlin/org/fbme/ide/integration/fordiac/lua/BasicFBTypeLuaTranslator.kt:20]: Why no just use `appendLine` where appropriate?

File: code/library/src/main/kotlin/org/fbme/lib/st/expressions/BinaryOperation.kt [hunk 2]
```diff
@@ -1,10 +1,10 @@
 package org.fbme.lib.st.expressions
 
-enum class BinaryOperation(val alias: String) {
-    ADD("+"), AMP("&"), AND("AND"), DIV("/"), EQ("="), GT(">"), GTE(">="), LT("<"), LTE("<="), MOD("MOD"), MUL("*"), NEQ(
-        "<>"
-    ),
-    OR("OR"), POW("**"), SUB("-"), XOR("XOR");
+enum class BinaryOperation(val alias: String, val luaAlias: String) {
```
Comment (qradimir) [inline code/library/src/main/kotlin/org/fbme/lib/st/expressions/BinaryOperation.kt:3]: I would extract `luaAlias` from this definition -- in general `library` code shall be not aware of specific runtimes.

File: code/library/src/main/kotlin/org/fbme/lib/st/expressions/UnaryOperation.kt [hunk 3]
```diff
@@ -1,7 +1,7 @@
 package org.fbme.lib.st.expressions
 
-enum class UnaryOperation(val alias: String, val isSpaced: Boolean) {
-    NOT("NOT", true), NEG("-", false);
+enum class UnaryOperation(val alias: String, val luaAlias: String, val isSpaced: Boolean) {
```
Comment (qradimir) [inline code/library/src/main/kotlin/org/fbme/lib/st/expressions/UnaryOperation.kt:3]: Same here, I would make a separate function in `4diac-integration` to map st operations to lua operations

**Review by qradimir — APPROVED** [review]

File: code/4diac-integration/src/main/kotlin/org/fbme/ide/integration/fordiac/lua/BasicFBTypeLuaTranslator.kt (hunk 1 above)
Comment (Ruslan227) [reply code/4diac-integration/src/main/kotlin/org/fbme/ide/integration/fordiac/lua/BasicFBTypeLuaTranslator.kt]: fixed

File: code/library/src/main/kotlin/org/fbme/lib/st/expressions/BinaryOperation.kt (hunk 2 above)
Comment (Ruslan227) [reply code/library/src/main/kotlin/org/fbme/lib/st/expressions/BinaryOperation.kt]: fixed

File: code/library/src/main/kotlin/org/fbme/lib/st/expressions/UnaryOperation.kt (hunk 3 above)
Comment (Ruslan227) [reply code/library/src/main/kotlin/org/fbme/lib/st/expressions/UnaryOperation.kt]: fixed

**Pull request merged**